### PR TITLE
[wasm][debugger] Stepping into hidden function.

### PR DESCRIPTION
**Pull request by ilonatommy**
How it works in Console App:

- Setting enabled breakpoints in the decorated method in VS is blocked. (DONE in Wasm)
- Stepping into the decorated method is blocked, instead debugger steps over. (DONE in Wasm)
- Using Debugger.Break() in the function result in pausing on the call of the function, no stepping in, not appearing on the call stack, no access to local variables. After resuming, the debugger steps over. (DONE in Wasm)
- Using Debugger.Break() in the function while we are stepping into results in 3 steps (in or over) on the method call. (A difference: in Blazor App it results in 2 steps on the method call).

### CODE REVIEW

File: src/mono/wasm/debugger/DebuggerTestSuite/SteppingTests.cs [hunk 1]
```diff
@@ -948,5 +948,25 @@ public async Task CheckResetFrameNumberForEachStep()
...
+                "window.setTimeout(function() { invoke_static_method('[debugger-test] DebuggerAttribute:Run'); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs",
+                pause_location.Value["locations"][0]["lineNumber"].Value<int>(),
+                pause_location.Value["locations"][0]["columnNumber"].Value<int>(),
+                "Run"
+            );
+            var step_into = await SendCommandAndCheck(null, $"Debugger.stepInto", null, -1, -1, null);
+
+            Assert.Equal(
+                step_into["callFrames"][0]["location"]["lineNumber"].Value<int>(),
```
Comment (thaystg) [inline src/mono/wasm/debugger/DebuggerTestSuite/SteppingTests.cs:966]: As we discussed offline, the step_into should be on the next line.

Comment (ilonatommy) [reply src/mono/wasm/debugger/DebuggerTestSuite/SteppingTests.cs]: I still need to correct the code so that the test will be passing.

**Comment by radical** [comment]
Would you mind describing the behavior of a hidden method in a debugger? For example, can you set breakpoints in it? Can you step into it from a visible method that you were already paused at?

In the `debugger.break` in the hidden method case, does it not show the callstack at all? As you are already in that method, can you step through the code? or step out? Can you access locals, evaluate expressions etc?

If you are paused on a visible method, but were called from a hidden method, then does that method not show up in the callstack? If it does, then can you inspect its locals?

Feel free to point me to a doc, if any, on this.

**Comment by ilonatommy** [comment]
@radical that's a good question. For now I added a short description to this PR but I will also prepare it in a form of doc.

**Review by thaystg — APPROVED** [review]

**Pull request merged**